ONSdigital/eq-survey-runner
# Improvements/Fixes to Gulp taks

### What is the context of this PR?

Styles compilation through Gulp was problematic, sometimes compiling twice and unnecessarily slow due to minification running in local environments.

Improvements made:
- updated all stylelint-related dependencies.
- moved css minification into gulp subtask, only invoked when env var `EQ_MINIMIZE_ASSETS` is `True` (ie. on pre/production environments).
- removed linting from `watch` task. (this should be handled in your editor).
- removed unused copying of webfonts.
- removed unnecessary scripts task from watch task.
### How to review
- `npm install` to install new dependencies.
- remove contents of `/static/` folder.
- `npm run compile` and verify app is still styled and functioning as expected (eg. accordion still works).
- remove contents of `/static/` folder again.
- `npm start` to kick off watching/browsersync.
- verify app works as expected.
- save a `scss` file, verify compilation/livereload work.
- save a `js` file, verify compilation/livereload work.
### Who worked on the PR

@hamishtaplin
